# LIVE CODE 3

<h1>MUHAMAD NATUAL HISAK</h1>

Kamu adalah seorang data analis di Austin Smart City. Saat ini, kamu sedang membantu kepolisian kota Austin untuk mempelajari kondisi tindak kejahatan di kota Austin.

Dataset yang akan digunakan dapat diakses menggunakan bigquery-public-data pada Google Cloud Big Query.

Supaya analisismu terarah, kita perlu membuat penjabaran masalah dan kemudian dapat disimpulkan. Namun, sebelum membuat penjabaran masalah, pastikan sudah menentukan problem statement agar kesimpulan yang dibuat lebih terarah.

1. Definisikan problem statement berdasarkan penjabaran masalah di bawah yang akan teman-teman jawab menggunakan SQL!

#Penjabaran Masalah
1. Tampilkan tabel crime sampai 5 baris pertama

2. Apa saja jenis tindak kejahatan yang terjadi di kota Austin selama tahun 2016?

3. Berapa banyak kasus tindak kejahatan berdasarkan kasus 
pembersihannya (clearance status) selama tahun 2016?

4. Jenis kejahatan apa saja yang paling banyak belum terselesaikan kasusnya di tahun 2016?

5. Di bulan apa di tahun 2016 kasus pencurian (theft) sangat banyak terjadi?

6. Di distrik apa yang paling banyak terjadi kasus pencurian selama 2016?

7. Berapa lama rata-rata waktu (dalam hari) kasus tindak kejahatan 'theft' terselesaikan sejak kasus tersebut terangkat selama tahun 2016? (Hint: Hitung selisih hari dari timestamp hingga clearance date)

# WORKING AREA

PROBLEM STATEMENT : 
**Mengetahui jumlah kasus dan lokasi distrik kejahatan berdasarkan jenis kasus Theft yang paling sering terjadi di Austin Smart City**

Import library pandas dan numpy

In [1]:
import pandas as pd
import numpy as np

Load dataset yang sudah di download dari bigquery dengan query : (SELECT * FROM `bigquery-public-data.austin_crime.crime`)

In [13]:
df = pd.read_csv(r'C:\Users\Omen\Desktop\Hacktiv8\Live Code 3\Austin Crime.csv')
df.head()

unique_key     address  census_tract           clearance_date  \
0  2.016500e+10  Austin, TX           NaN  2016-03-14 12:00:00 UTC   
1  2.016500e+10  Austin, TX           NaN  2016-03-31 12:00:00 UTC   
2  2.016251e+09  Austin, TX           NaN  2016-09-16 12:00:00 UTC   
3  2.016500e+10  Austin, TX           NaN  2016-02-09 12:00:00 UTC   
4  2.016322e+09  Austin, TX           NaN  2016-03-22 12:00:00 UTC   

  clearance_status  council_district_code                     description  \
0      Not cleared                    NaN  RAPE                             
1      Not cleared                    NaN  RAPE                             
2      Not cleared                    NaN  RAPE                             
3      Not cleared                    NaN  AGG RAPE OF A CHILD              
4      Not cleared                    NaN  RAPE OF A CHILD                  

  district  latitude  longitude location location_description primary_type  \
0       UK       NaN        NaN      NaN                  NaN         Rape   
1       UK       NaN        NaN      NaN                  NaN         Rape   
2       UK       NaN        NaN      NaN                  NaN         Rape   
3       UK       NaN        NaN      NaN                  NaN         Rape   
4       UK       NaN        NaN      NaN                  NaN         Rape   

                 timestamp  x_coordinate  y_coordinate  year  zipcode  
0  2016-01-19 12:00:00 UTC           NaN           NaN  2016      NaN  
1  2016-01-25 12:00:00 UTC           NaN           NaN  2016      NaN  
2  2016-01-25 12:00:00 UTC           NaN           NaN  2016      NaN  
3  2016-01-27 12:00:00 UTC           NaN           NaN  2016      NaN  
4  2016-02-01 12:00:00 UTC           NaN           NaN  2016      NaN

Cek missing value

In [14]:
df.info(null_counts= True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 116675 entries, 0 to 116674
Data columns (total 18 columns):
 #   Column                 Non-Null Count   Dtype  
---  ------                 --------------   -----  
 0   unique_key             116674 non-null  float64
 1   address                116675 non-null  object 
 2   census_tract           116152 non-null  float64
 3   clearance_date         111561 non-null  object 
 4   clearance_status       111561 non-null  object 
 5   council_district_code  116000 non-null  float64
 6   description            116675 non-null  object 
 7   district               116675 non-null  object 
 8   latitude               34500 non-null   float64
 9   longitude              34500 non-null   float64
 10  location               34500 non-null   object 
 11  location_description   114874 non-null  object 
 12  primary_type           116675 non-null  object 
 13  timestamp              116675 non-null  object 
 14  x_coordinate           114528 non-nu

C:\Users\Omen\AppData\Roaming\Python\Python37\site-packages\ipykernel_launcher.py:1: FutureWarning: null_counts is deprecated. Use show_counts instead
  """Entry point for launching an IPython kernel.


Terlihat bahwa banyak missing value pada tabel diatas

<h2>DATA CLEANING</h2>

Copy tabel ke variabel baru agar tidak terjadi overwrite pada saat proses data cleaning

In [15]:
x = df.copy()
x.head()

unique_key     address  census_tract           clearance_date  \
0  2.016500e+10  Austin, TX           NaN  2016-03-14 12:00:00 UTC   
1  2.016500e+10  Austin, TX           NaN  2016-03-31 12:00:00 UTC   
2  2.016251e+09  Austin, TX           NaN  2016-09-16 12:00:00 UTC   
3  2.016500e+10  Austin, TX           NaN  2016-02-09 12:00:00 UTC   
4  2.016322e+09  Austin, TX           NaN  2016-03-22 12:00:00 UTC   

  clearance_status  council_district_code                     description  \
0      Not cleared                    NaN  RAPE                             
1      Not cleared                    NaN  RAPE                             
2      Not cleared                    NaN  RAPE                             
3      Not cleared                    NaN  AGG RAPE OF A CHILD              
4      Not cleared                    NaN  RAPE OF A CHILD                  

  district  latitude  longitude location location_description primary_type  \
0       UK       NaN        NaN      NaN                  NaN         Rape   
1       UK       NaN        NaN      NaN                  NaN         Rape   
2       UK       NaN        NaN      NaN                  NaN         Rape   
3       UK       NaN        NaN      NaN                  NaN         Rape   
4       UK       NaN        NaN      NaN                  NaN         Rape   

                 timestamp  x_coordinate  y_coordinate  year  zipcode  
0  2016-01-19 12:00:00 UTC           NaN           NaN  2016      NaN  
1  2016-01-25 12:00:00 UTC           NaN           NaN  2016      NaN  
2  2016-01-25 12:00:00 UTC           NaN           NaN  2016      NaN  
3  2016-01-27 12:00:00 UTC           NaN           NaN  2016      NaN  
4  2016-02-01 12:00:00 UTC           NaN           NaN  2016      NaN

Hapus kolom yang tidak diperlukan

In [17]:
x = x.drop(["latitude", "longitude","census_tract","council_district_code","location","zipcode"], axis = 1)
x

unique_key                                            address  \
0       2.016500e+10                                         Austin, TX   
1       2.016500e+10                                         Austin, TX   
2       2.016251e+09                                         Austin, TX   
3       2.016500e+10                                         Austin, TX   
4       2.016322e+09                                         Austin, TX   
...              ...                                                ...   
116670  2.014800e+10  13813 N MOPAC EXPY NB Austin, TX 78727 (30.439...   
116671  2.014502e+10  14205 N MOPAC EXPY NB Austin, TX 78727 (30.444...   
116672  2.014503e+10  12761 COUNCIL BLUFF DR Austin, TX 78727 (30.42...   
116673  2.014503e+10  12805 COUNCIL BLUFF DR Austin, TX 78727 (30.42...   
116674  2.014506e+10  13305 COUNCIL BLUFF DR Austin, TX 78727 (30.43...   

                 clearance_date clearance_status  \
0       2016-03-14 12:00:00 UTC      Not cleared   
1       2016-03-31 12:00:00 UTC      Not cleared   
2       2016-09-16 12:00:00 UTC      Not cleared   
3       2016-02-09 12:00:00 UTC      Not cleared   
4       2016-03-22 12:00:00 UTC      Not cleared   
...                         ...              ...   
116670  2014-03-05 12:00:00 UTC      Not cleared   
116671  2014-06-02 12:00:00 UTC      Not cleared   
116672  2014-07-31 12:00:00 UTC      Not cleared   
116673  2014-07-24 12:00:00 UTC      Not cleared   
116674  2015-01-02 12:00:00 UTC      Not cleared   

                           description district    location_description  \
0       RAPE                                 UK                     NaN   
1       RAPE                                 UK                     NaN   
2       RAPE                                 UK                     NaN   
3       AGG RAPE OF A CHILD                  UK                     NaN   
4       RAPE OF A CHILD                      UK                     NaN   
...                                ...      ...                     ...   
116670  BURGLARY OF VEHICLE                   A   13813 N MOPAC EXPY NB   
116671  THEFT                                 A   14205 N MOPAC EXPY NB   
116672  THEFT                                 A  12761 COUNCIL BLUFF DR   
116673  THEFT OF AUTO PARTS                   A  12805 COUNCIL BLUFF DR   
116674  THEFT                                 A  13305 COUNCIL BLUFF DR   

                    primary_type                timestamp  x_coordinate  \
0                           Rape  2016-01-19 12:00:00 UTC           NaN   
1                           Rape  2016-01-25 12:00:00 UTC           NaN   
2                           Rape  2016-01-25 12:00:00 UTC           NaN   
3                           Rape  2016-01-27 12:00:00 UTC           NaN   
4                           Rape  2016-02-01 12:00:00 UTC           NaN   
...                          ...                      ...           ...   
116670                Theft: BOV  2014-03-03 12:00:00 UTC     3127011.0   
116671  Theft: All Other Larceny  2014-05-30 12:00:00 UTC     3127296.0   
116672  Theft: All Other Larceny  2014-07-29 12:00:00 UTC     3120803.0   
116673         Theft: Auto Parts  2014-07-23 12:00:00 UTC     3120717.0   
116674  Theft: All Other Larceny  2014-12-26 12:00:00 UTC     3121653.0   

        y_coordinate  year  
0                NaN  2016  
1                NaN  2016  
2                NaN  2016  
3                NaN  2016  
4                NaN  2016  
...              ...   ...  
116670    10133852.0  2014  
116671    10134835.0  2014  
116672    10129736.0  2014  
116673    10129781.0  2014  
116674    10131803.0  2014  

[116675 rows x 12 columns]

Cek missing value lagi 

In [19]:
x.info(null_counts = True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 116675 entries, 0 to 116674
Data columns (total 12 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   unique_key            116674 non-null  float64
 1   address               116675 non-null  object 
 2   clearance_date        111561 non-null  object 
 3   clearance_status      111561 non-null  object 
 4   description           116675 non-null  object 
 5   district              116675 non-null  object 
 6   location_description  114874 non-null  object 
 7   primary_type          116675 non-null  object 
 8   timestamp             116675 non-null  object 
 9   x_coordinate          114528 non-null  float64
 10  y_coordinate          114528 non-null  float64
 11  year                  116675 non-null  int64  
dtypes: float64(3), int64(1), object(8)
memory usage: 10.7+ MB


C:\Users\Omen\AppData\Roaming\Python\Python37\site-packages\ipykernel_launcher.py:1: FutureWarning: null_counts is deprecated. Use show_counts instead
  """Entry point for launching an IPython kernel.


Untuk menghandle missing value pada tabel baru ini, pada kolom clearance_date dan clearance_status akan diisi "NOT CLEAR" karena diasumsikan bahwa kasus tersebut belum selesai.<br>
pada kolom x_coordinate, y_coordinate, dan location_description akan diisi "UNKNOWN" karena tidak ada data yang menunjukkan koordinat dan lokasi. <br>
drop satu missing value yang ada pada kolom unique_key


Ganti type data pada kolom x dan y coordinate

In [25]:
x['x_coordinate'] = df['x_coordinate'].astype(object)
x['y_coordinate'] = df['y_coordinate'].astype(object)
x.info(null_counts=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 116675 entries, 0 to 116674
Data columns (total 12 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   unique_key            116674 non-null  float64
 1   address               116675 non-null  object 
 2   clearance_date        111561 non-null  object 
 3   clearance_status      111561 non-null  object 
 4   description           116675 non-null  object 
 5   district              116675 non-null  object 
 6   location_description  114874 non-null  object 
 7   primary_type          116675 non-null  object 
 8   timestamp             116675 non-null  object 
 9   x_coordinate          114528 non-null  object 
 10  y_coordinate          114528 non-null  object 
 11  year                  116675 non-null  int64  
dtypes: float64(1), int64(1), object(10)
memory usage: 10.7+ MB


C:\Users\Omen\AppData\Roaming\Python\Python37\site-packages\ipykernel_launcher.py:3: FutureWarning: null_counts is deprecated. Use show_counts instead
  This is separate from the ipykernel package so we can avoid doing imports until


Isi missing value sesuai petunjuk diatas

In [26]:
values = {"clearance_date":"NOT CLEAR", "clearance_status":"NOT CLEAR","x_coordinate":"Unknown","y_coordinate":"Unknown","location_description":"Unknown"}
x = x.fillna(value=values)
x

unique_key                                            address  \
0       2.016500e+10                                         Austin, TX   
1       2.016500e+10                                         Austin, TX   
2       2.016251e+09                                         Austin, TX   
3       2.016500e+10                                         Austin, TX   
4       2.016322e+09                                         Austin, TX   
...              ...                                                ...   
116670  2.014800e+10  13813 N MOPAC EXPY NB Austin, TX 78727 (30.439...   
116671  2.014502e+10  14205 N MOPAC EXPY NB Austin, TX 78727 (30.444...   
116672  2.014503e+10  12761 COUNCIL BLUFF DR Austin, TX 78727 (30.42...   
116673  2.014503e+10  12805 COUNCIL BLUFF DR Austin, TX 78727 (30.42...   
116674  2.014506e+10  13305 COUNCIL BLUFF DR Austin, TX 78727 (30.43...   

                 clearance_date clearance_status  \
0       2016-03-14 12:00:00 UTC      Not cleared   
1       2016-03-31 12:00:00 UTC      Not cleared   
2       2016-09-16 12:00:00 UTC      Not cleared   
3       2016-02-09 12:00:00 UTC      Not cleared   
4       2016-03-22 12:00:00 UTC      Not cleared   
...                         ...              ...   
116670  2014-03-05 12:00:00 UTC      Not cleared   
116671  2014-06-02 12:00:00 UTC      Not cleared   
116672  2014-07-31 12:00:00 UTC      Not cleared   
116673  2014-07-24 12:00:00 UTC      Not cleared   
116674  2015-01-02 12:00:00 UTC      Not cleared   

                           description district    location_description  \
0       RAPE                                 UK                 Unknown   
1       RAPE                                 UK                 Unknown   
2       RAPE                                 UK                 Unknown   
3       AGG RAPE OF A CHILD                  UK                 Unknown   
4       RAPE OF A CHILD                      UK                 Unknown   
...                                ...      ...                     ...   
116670  BURGLARY OF VEHICLE                   A   13813 N MOPAC EXPY NB   
116671  THEFT                                 A   14205 N MOPAC EXPY NB   
116672  THEFT                                 A  12761 COUNCIL BLUFF DR   
116673  THEFT OF AUTO PARTS                   A  12805 COUNCIL BLUFF DR   
116674  THEFT                                 A  13305 COUNCIL BLUFF DR   

                    primary_type                timestamp x_coordinate  \
0                           Rape  2016-01-19 12:00:00 UTC      Unknown   
1                           Rape  2016-01-25 12:00:00 UTC      Unknown   
2                           Rape  2016-01-25 12:00:00 UTC      Unknown   
3                           Rape  2016-01-27 12:00:00 UTC      Unknown   
4                           Rape  2016-02-01 12:00:00 UTC      Unknown   
...                          ...                      ...          ...   
116670                Theft: BOV  2014-03-03 12:00:00 UTC    3127011.0   
116671  Theft: All Other Larceny  2014-05-30 12:00:00 UTC    3127296.0   
116672  Theft: All Other Larceny  2014-07-29 12:00:00 UTC    3120803.0   
116673         Theft: Auto Parts  2014-07-23 12:00:00 UTC    3120717.0   
116674  Theft: All Other Larceny  2014-12-26 12:00:00 UTC    3121653.0   

       y_coordinate  year  
0           Unknown  2016  
1           Unknown  2016  
2           Unknown  2016  
3           Unknown  2016  
4           Unknown  2016  
...             ...   ...  
116670   10133852.0  2014  
116671   10134835.0  2014  
116672   10129736.0  2014  
116673   10129781.0  2014  
116674   10131803.0  2014  

[116675 rows x 12 columns]

Drop missing value pada unique_key lalu masukkan tabel ke variabel baru

In [30]:
xc = x.dropna()

Cek missing value

In [31]:
xc.info(null_counts=True)

<class 'pandas.core.frame.DataFrame'>
Int64Index: 116674 entries, 0 to 116674
Data columns (total 12 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   unique_key            116674 non-null  float64
 1   address               116674 non-null  object 
 2   clearance_date        116674 non-null  object 
 3   clearance_status      116674 non-null  object 
 4   description           116674 non-null  object 
 5   district              116674 non-null  object 
 6   location_description  116674 non-null  object 
 7   primary_type          116674 non-null  object 
 8   timestamp             116674 non-null  object 
 9   x_coordinate          116674 non-null  object 
 10  y_coordinate          116674 non-null  object 
 11  year                  116674 non-null  int64  
dtypes: float64(1), int64(1), object(10)
memory usage: 11.6+ MB


C:\Users\Omen\AppData\Roaming\Python\Python37\site-packages\ipykernel_launcher.py:1: FutureWarning: null_counts is deprecated. Use show_counts instead
  """Entry point for launching an IPython kernel.


Sudah tidak ada missing value dan data siap di eksplorasi 

<h2>PENJABARAN MASALAH</h2>

1. Tampilkan tabel crime sampai 5 baris pertama

Query : 
```
SELECT * FROM `bigquery-public-data.austin_crime.crime` limit 5
```

In [5]:
df1 = pd.read_csv(r'C:\Users\Omen\Desktop\Hacktiv8\Live Code 3\P0LC3_Muhamad Natual Hisak_1.csv')
df1

unique_key     address  census_tract                  clearance_date  \
0  20165002898  Austin, TX           NaN  2016-03-14 12:00:00.000000 UTC   
1  20165004843  Austin, TX           NaN  2016-03-31 12:00:00.000000 UTC   
2   2016250988  Austin, TX           NaN  2016-09-16 12:00:00.000000 UTC   
3  20165003839  Austin, TX           NaN  2016-02-09 12:00:00.000000 UTC   
4   2016321746  Austin, TX           NaN  2016-03-22 12:00:00.000000 UTC   

  clearance_status  council_district_code                     description  \
0      Not cleared                    NaN  RAPE                             
1      Not cleared                    NaN  RAPE                             
2      Not cleared                    NaN  RAPE                             
3      Not cleared                    NaN  AGG RAPE OF A CHILD              
4      Not cleared                    NaN  RAPE OF A CHILD                  

  district  latitude  longitude  location  location_description primary_type  \
0       UK       NaN        NaN       NaN                   NaN         Rape   
1       UK       NaN        NaN       NaN                   NaN         Rape   
2       UK       NaN        NaN       NaN                   NaN         Rape   
3       UK       NaN        NaN       NaN                   NaN         Rape   
4       UK       NaN        NaN       NaN                   NaN         Rape   

                        timestamp  x_coordinate  y_coordinate  year  zipcode  
0  2016-01-19 12:00:00.000000 UTC           NaN           NaN  2016      NaN  
1  2016-01-25 12:00:00.000000 UTC           NaN           NaN  2016      NaN  
2  2016-01-25 12:00:00.000000 UTC           NaN           NaN  2016      NaN  
3  2016-01-27 12:00:00.000000 UTC           NaN           NaN  2016      NaN  
4  2016-02-01 12:00:00.000000 UTC           NaN           NaN  2016      NaN

2. Apa saja jenis tindak kejahatan yang terjadi di kota Austin selama tahun 2016?

Query : 
```
SELECT extract(year from timestamp) as year,
       primary_type, 
       count(unique_key) as total
FROM `bigquery-public-data.austin_crime.crime` 
where timestamp >= "2016-01-01" and timestamp <= "2016-12-31"
GROUP BY 1,2
```

In [6]:
df2 = pd.read_csv(r'C:\Users\Omen\Desktop\Hacktiv8\Live Code 3\P0LC3_Muhamad Natual Hisak_2.csv')
df2

year primary_type  total
0  2016         Rape    755
1  2016        Theft  26191
2  2016      Robbery   1047
3  2016  Agg Assault   2172
4  2016     Burglary   5161
5  2016   Auto Theft   2021
6  2016       Murder     38

Bisa dilihat bahwa jenis tindak kejahatan yang terjadi di kota Austin selama tahun 2016 ada 7 jenis dengan "Theft" sebagai jenis kejahatan yang paling banyak terjadi dengan selisih yang sangat jauh sekali dengan jenis-jenis kejahatan lainnya.

3. Berapa banyak kasus tindak kejahatan berdasarkan kasus pembersihannya (clearance status) selama tahun 2016?

Query : 
```
SELECT extract(year from timestamp) as year,
       clearance_status, 
       count(unique_key) as total
FROM `bigquery-public-data.austin_crime.crime` 
where timestamp >= "2016-01-01" and timestamp <= "2016-12-31"
GROUP BY 1,2
```

In [7]:
df3 = pd.read_csv(r'C:\Users\Omen\Desktop\Hacktiv8\Live Code 3\P0LC3_Muhamad Natual Hisak_3.csv')
df3

year      clearance_status  total
0  2016           Not cleared  29664
1  2016  Cleared by Exception   1012
2  2016                   NaN   1711
3  2016     Cleared by Arrest   4998

Bisa dilihat bahwa di Kota Austin pada tahun 2016, kasus kejahatan ditanggulangi dengan damai bersyarat (cleared by exception), penahanan (arrest), tidak atau belum selesai, dan beberapa kasus yang tidak diketahui hasilnya. kasus kejahatan yang tidak atau belum diselesaikan jumlahnya sangat-sangat banyak dibandingkan dengan status kasus yang lain.

4. Jenis kejahatan apa saja yang paling banyak belum terselesaikan kasusnya di tahun 2016?

Query : 
```
SELECT extract(year from timestamp) as year,primary_type, count(unique_key) as total
FROM `bigquery-public-data.austin_crime.crime` 
where timestamp >= "2016-01-01" and timestamp <= "2016-12-31" and clearance_status = "Not cleared"
GROUP BY 1,2
ORDER BY 3 desc

```

In [8]:
df4 = pd.read_csv(r'C:\Users\Omen\Desktop\Hacktiv8\Live Code 3\P0LC3_Muhamad Natual Hisak_4.csv')
df4

year primary_type  total
0  2016        Theft  21860
1  2016     Burglary   4432
2  2016   Auto Theft   1537
3  2016  Agg Assault    915
4  2016      Robbery    559
5  2016         Rape    360
6  2016       Murder      1

Dari data diatas, bisa dilihat bahwa kasus yang paling banyak belum diselesaikan di Kota Austin pada tahun 2016 adalah Theft dengan lagi-lagi jumlah yang sangat jauh dibandingkan jenis kejahatan lainnya.

5. Di bulan apa di tahun 2016 kasus pencurian (theft) sangat banyak terjadi?

Query :
```
SELECT extract(year from timestamp) as year, extract(month from timestamp) as month,primary_type, count(unique_key) as total
FROM `bigquery-public-data.austin_crime.crime` 
where timestamp >= "2016-01-01" and timestamp <= "2016-12-31" and primary_type = "Theft"
GROUP BY 1,2,3
ORDER BY 4 desc

```

In [9]:
df5 = pd.read_csv(r'C:\Users\Omen\Desktop\Hacktiv8\Live Code 3\P0LC3_Muhamad Natual Hisak_5.csv')
df5

year  month primary_type  total
0   2016      4        Theft   2282
1   2016      3        Theft   2271
2   2016      6        Theft   2249
3   2016     10        Theft   2235
4   2016      1        Theft   2213
5   2016      8        Theft   2204
6   2016      5        Theft   2183
7   2016      7        Theft   2163
8   2016      9        Theft   2156
9   2016     11        Theft   2128
10  2016     12        Theft   2113
11  2016      2        Theft   1994

Pada tahun 2016, di kota Austin paling banyak terjadi kejahatan dengan tipe Theft ada pada bulan ke-4 atau bulan April. Bisa dilihat juga dari data diatas bahwa jumlah terjadinya kejahatan tipe theft pada tahun 2016 ini tersebar merata dari bulan ke bulan jadi tidak ada satu bulan yang jumlah kejahatan tipe theft ini menonjol sendiri.

6. Di distrik apa yang paling banyak terjadi kasus pencurian selama 2016?

Query : 
```
SELECT extract(year from timestamp) as year,district, count(unique_key) as total
FROM `bigquery-public-data.austin_crime.crime` 
where timestamp >= "2016-01-01" and timestamp <= "2016-12-31" and primary_type = "Theft"
GROUP BY 1,2
ORDER BY  3 desc

```

In [32]:
df6 = pd.read_csv(r'C:\Users\Omen\Desktop\Hacktiv8\Live Code 3\P0LC3_Muhamad Natual Hisak_6.csv')
df6

year district  total
0   2016        D   4108
1   2016        B   3324
2   2016        A   3093
3   2016        F   3031
4   2016        E   2976
5   2016        I   2864
6   2016        H   2726
7   2016        G   2112
8   2016        C   1722
9   2016       AP    143
10  2016       UK     79
11  2016       88     13

Dilihat dari data diatas, distrik kota Austin yang paling banyak terjadi kasus pencurian (Theft) pada tahun 2016 adalah distrik D.

7. Berapa lama rata-rata waktu (dalam hari) kasus tindak kejahatan 'theft' terselesaikan sejak kasus tersebut terangkat selama tahun 2016? (Hint: Hitung selisih hari dari timestamp hingga clearance date)

Query : 

```
SELECT AVG(days) as average_days 
FROM(
    SELECT day_cleared,day_reported,
           DATE_DIFF(day_cleared, day_reported,day) as days 
    FROM (
      SELECT
        timestamp as day_reported,
        clearance_date as day_cleared
      FROM `bigquery-public-data.austin_crime.crime` 
      where timestamp >= "2016-01-01" and timestamp <= "2016-12-31" and primary_type = "Theft"
      GROUP BY 1,2))

```

In [33]:
df7 = pd.read_csv(r'C:\Users\Omen\Desktop\Hacktiv8\Live Code 3\P0LC3_Muhamad Natual Hisak_7.csv')
df7

average_days
0     31.905987

Rata-rata waktu untuk menyelesaikan kasus kejahatan tipe Theft pada kota Austin di tahun 2016 adalah 31.9 hari yang berarti waktu yang dibutuhkan adalah sedikit lebih dari 1 bulan.

# Poin Analisis

1. Dari penjabaran yang sudah kamu bahas (pertanyaan 2-7), apa kesimpulan yang kamu dapat sampaikan?

```
Kesimpulan yang dapat saya sampaikan adalah, Pada Kota Austin di tahun 2016, kejahatan yang terjadi ada 7 Jenis yaitu :
1. Theft
2. Robbery	
3. Rape
4. Murder
5. Agg Assault
6. Burglary
7. Auto Theft

Dari ke 7 jenis kejahatan tersebut. yang paling sering terjadi adalah Theft.
Kasus terbanyak yang belum selesai di tindak juga ada pada kategori Theft.
Yang menjadi waktu terbanyak terjadinya kasus Theft adalah bulan April.
Distrik yang paling sering terjadi kasus Theft adalah distrik D.
Rata-rata waktu untuk menyelesaikan kasus Theft adalah 31.9 Hari atau kurang lebih 1 bulan.


```